sql-access-layer/sql-access-layer/src/main/java/com/cancerup/sqlaccesslayer/controller/EventController.java

```diff
@@ -0,0 +1,34 @@
+package com.cancerup.sqlaccesslayer.controller;
+import com.cancerup.sqlaccesslayer.EventRepository;
+import com.cancerup.sqlaccesslayer.models.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Optional;
+
+@RestController
+public class EventController {
+
+    @Autowired
+    private EventRepository eventRepository;
+
+    @RequestMapping(value="/createevent", method= RequestMethod.POST)
+    public ResponseEntity<Void> createEvent(@RequestBody Event event)  {
+        if (eventRepository.findByEventId(event.getEventId()).equals(Optional.empty())) {
```

Comment:
```suggestion
        if (event.userId && event.eventName) //If either of these two things are null it will fail
```
This will never be false since the event ID is set to auto-increment. I don't believe there should ever be a collision in the events table. The only check we should do before adding it to the database is to check that the user_id exists and that we have an event_name in the object we are trying to save.  This way we ensure every event is tied to a user and there is at minimum an event name. Also, I am testing the code review functionality so let me know how this goes!